state/factoidstate.go

```diff
@@ -324,14 +325,25 @@ func (fs *FactoidState) UpdateTransaction(rt bool, trans interfaces.ITransaction
 	for _, output := range trans.GetOutputs() {
 		adr := output.GetAddress().Fixed()
 		oldv := fs.State.GetF(rt, adr)
+
+		fs.State.LogPrintf("newHolding", "process FCT Deposit %x %s", adr, trans.String())
+		fs.State.ExecuteFromHolding(adr) // Process deposit of FCT
+
 		fs.State.PutF(rt, adr, oldv+int64(output.GetAmount()))
 	}
 	if len(trans.GetECOutputs()) > 0 {
 		fs.State.LogPrintf("entrycredits", "At %d process %s", fs.DBHeight, trans.String())
 	}
 	for _, ecOut := range trans.GetECOutputs() {
 		ecbal := int64(ecOut.GetAmount()) / int64(fs.State.FactoshisPerEC)
-		fs.State.PutE(rt, ecOut.GetAddress().Fixed(), fs.State.GetE(rt, ecOut.GetAddress().Fixed())+ecbal)
+		adr := ecOut.GetAddress().Fixed()
+		fs.State.PutE(rt, adr, fs.State.GetE(rt, adr)+ecbal) // Add EC's from FCT
+
+		// execute any messages that were waiting on this EC address
+		if rt == true {
```

Comment:
Is there a reason why we check if rt is true here, but not for the fct outputs?

Reply:
we spoke about this - commenting for posterity
w/ the condition in place - only changes to temporary balance would result in triggering an execution of held messages
we ended up removing this so execution will be attempted for any balance change

this is to account for an apparent edge case where the balance changes because the node loaded a Db state rather then processing messages